### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by jonschlinkert**
@phated / @doowb, how's this? It uses fewer libraries and only iterates over the value once. If you're good with this I'll merge and publish as 1.0.0.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
-  if (isObject(collection) && !collection.length) {
-    collection = values(collection);
+module.exports = function(collection, fn, thisArg) {
+  if (!Array.isArray(collection)) {
+    var iterator = makeIterator(fn);
+    var result = [];
+
+    forOwn(collection, function(value, key) {
+      if (typeof fn === 'string') {
+        console.log(collection[fn])
```
Comment (doowb) [inline index.js]: looks like this was left in by mistake

**Comment by doowb** [comment]
I like this better since you have the option of getting the `key` and `value` in the function if you need it.

There was one line I made a comment on that I think was just left in by mistake, but it looks good.

File: index.js (hunk 1 above)
Comment (jonschlinkert) [reply index.js]: oh I thought you'd like for me to leave some Easter eggs in the code. jk lol. thx I got distracted when my girls got home earlier.

**Comment by jonschlinkert** [comment]
I'm going to go ahead and merge this, it makes more sense than the current code, but I can make changes if necessary.

**Pull request merged**